### PR TITLE
IGNITE-23944 Sql. Add TPC DS suite

### PR DESCRIPTION
**Pull request by korlov42**
https://issues.apache.org/jira/browse/IGNITE-23944

-----------
Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpcdsBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+@State(Scope.Benchmark)
+@Fork(1)
+@Threads(1)
+@Warmup(iterations = 10, time = 2)
+@Measurement(iterations = 20, time = 2)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@SuppressWarnings({"WeakerAccess", "unused"})
+public class TpcdsBenchmark extends AbstractTpcBenchmark {
+    /*
```
Comment (ygerzhedovich) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpcdsBenchmark.java:55]: Javadoc is wrong

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpcdsBenchmark.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+/**
+ * Benchmark that runs sql queries from TPC-H suite via embedded client.
```
Comment (ygerzhedovich) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpcdsBenchmark.java:44]: need to amend

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpcdsBenchmark.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        try {
+            queryString = TpcdsHelper.getQuery(queryId);
+        } catch (Throwable e) {
+            nodeTearDown();
+
+            throw e;
+        }
+    }
+
+    /** Benchmark that measures performance of queries from TPC-H suite. */
```
Comment (ygerzhedovich) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpcdsBenchmark.java:94]: need to check all )

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpchBenchmark.java [hunk 4]
```diff
@@ -54,7 +51,7 @@
 @BenchmarkMode(Mode.AverageTime)
 @OutputTimeUnit(TimeUnit.MILLISECONDS)
 @SuppressWarnings({"WeakerAccess", "unused"})
-public class TpchBenchmark extends AbstractMultiNodeBenchmark {
+public class TpchBenchmark extends AbstractTpcBenchmark {
```
Comment (ygerzhedovich) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpchBenchmark.java:54]: please fix javadoc below, due to fsync mode always is off from now

File: modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/TpcTable.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    /** Returns name of the column with given 0-based index. */
+    String columnName(int idx);
+
+    /** Returns definition of a table including necessary indexes. */
+    String ddlScript();
+
+    /**
+     * Returns DML string representing single-row INSERT statement with dynamic parameters placeholders.
+     *
+     * <p>The order of columns matches the order specified in TPC-H specification
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/TpcTable.java:43]: TPC-*

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpcdsBenchmark.java (hunk 1 above)
Comment (korlov42) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpcdsBenchmark.java:55]: done

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpcdsBenchmark.java (hunk 2 above)
Comment (korlov42) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpcdsBenchmark.java]: done

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpcdsBenchmark.java (hunk 3 above)
Comment (korlov42) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpcdsBenchmark.java]: done

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpchBenchmark.java (hunk 4 above)
Comment (korlov42) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TpchBenchmark.java:54]: done

File: modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/TpcTable.java (hunk 5 above)
Comment (korlov42) [reply modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/TpcTable.java]: done

**Review by ygerzhedovich — APPROVED** [review]

**Pull request merged**